tests/unit/test_bugjira.py

```diff
@@ -180,6 +182,94 @@ def test_get_issue_with_non_string_key(sandboxed_bugjira):
         sandboxed_bugjira.get_issue(1)
 
 
+def test_add_comment_non_issue(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance
+    WHEN we call add_comment with a None value for the 'issue' parameter
+    THEN a ValueError should be raised
+    """
+    with pytest.raises(ValueError,
+                       match="issue must be an Issue"):
+        sandboxed_bugjira.add_comment(None, "")
+
+
+def test_add_comment_non_string_comment(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance
+    WHEN we call add_comment with a None value for the 'comment' parameter
+    THEN a ValueError should be raised
+    """
+    with pytest.raises(ValueError,
+                       match="comment must be a str"):
+        sandboxed_bugjira.add_comment(Issue(key="foo"),
+                                      None)
+
+
+def test_add_comment_good_bugzilla(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance
+    WHEN we call add_comment with a BugzillaIssue and comment string
+    THEN the bugzilla backend's build_update and update_bugs methods should
+        each be invoked once
+    """
+    comment_text = "This is a sample comment"
+    key = "123456"
+    issue = BugzillaIssue(key=key)
+    sandboxed_bugjira.add_comment(issue, comment_text)
+    assert sandboxed_bugjira.bugzilla.build_update.call_count == 1
+    assert sandboxed_bugjira.bugzilla.update_bugs.call_count == 1
+
+
+def test_add_comment_bugzilla_exception(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance whose bugzilla backend's update_bugs method
+        raises an Exception
+    WHEN we call add_comment
+    THEN a BrokerAddCommentException should be raised
+    """
+    def raise_exception(*args):
+        raise Exception("Exception raised")
+    sandboxed_bugjira.bugzilla.update_bugs = raise_exception
+    comment_text = "This is a sample comment"
+    key = "123456"
+    issue = BugzillaIssue(key=key)
+    with pytest.raises(BrokerAddCommentException,
+                       match="Exception raised"):
+        sandboxed_bugjira.add_comment(issue, comment_text)
+
+
+def test_add_comment_good_jira(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance
+    WHEN we call add_comment with a JiraIssue and comment string
+    THEN the bugzilla backend's build_update and update_bugs methods should
```

Comment:
build_update doesn't seem to be checked here, either the docstring or the test should be tweaked.

Reply:
thank you for noticing my bad copy/paste!

Reply:
Always happy to be of service ;D